### PR TITLE
Remove dataloader state dict loading

### PR DESCRIPTION
**Pull request by felipemello1**
#### Context
What is the purpose of this PR? Is it to
- [ ] add a new feature
- [x] fix a bug
- [ ] update tests and/or documentation
- [ ] other (please add here)

Our current dataloader is epoch based. We do not need to load the state dict, as long as we call 'set_epoch', which sets the seed to the epoch number + original seed:
```
for curr_epoch in range(self.epochs_run, self.total_epochs):
            self._dataloader.sampler.set_epoch(curr_epoch)
```

The current implementation loads the state dict, then does a full pass on the dataloader, i.e. list(dataloader). For epoch based ckpt, this doesnt help, but make resuming extremely slow, since you have to go over all of it. 

For step based ckpt, we do NOT need list(dataloader). We just need to load the ckpt. 

NOTE: Loading ckpt state dict will need to be reintroduced when we change to step based ckpt.

#### Test plan

unit tests + running multimodal. Notice that grads are not the same, but what matters for this PR is that the batches are the same, and they are, given the loss shape.

```
tune run --nproc_per_node 2 lora_finetune_distributed --config llama3_2_vision/11B_lora epochs=2 max_steps_per_epoch=20 metric_logger=torchtune.training.metric_logging.WandBLogger resume_from_checkpoint=True checkpointer.adapter_checkpoint="epoch_0/adapter_model.pt"
```

image

```
tune run full_finetune_single_device --config llama3_2/1B_full_single_device epochs=2 max_steps_per_epoch=20 optimizer=torch.optim.SGDresume_from_checkpoint=True checkpointer.checkpoint_files=["epoch_0/model-00001-of-00001.safetensors"] 
```
image

### CODE REVIEW

**Comment by felipemello1** [comment]
cc: @krammnic @ankitageorge @ebsmothers @joecummings @pbontrager

**Comment by krammnic** [comment]
@felipemello1 Looks like I need to spread this on #2431

**Comment by felipemello1** [comment]
> @felipemello1 Looks like I need to spread this on #2431

I guess you meant #2441? yeah, thats why I tagged  you :D

**Review by ebsmothers — APPROVED** [review]
